### ISSUE TITLE
Front-end seems to be waiting for an API located at localhost:8000 even if parametrized otherwise in the compose file

### ISSUE THREAD

**Issue by swiss-knight** [taigaio/taiga-docker#23]
**Describe the bug**
The front-end seems to be waiting for an API located at `localhost:8000` even if I set up `my-lan-server.domain.org:9000` in the docker-compose.yml.

**To Reproduce**
- cloned the repo using git && cd taiga-docker
- edited the `docker-compose.yml` and `docker-compose-init.yml` to suit the needs of a remote server installation (inside a LAN).
A proof that it got propagated to the `taiga-back` container:
```
    # echo $TAIGA_SITES_DOMAIN
    my-lan-server.domain.org:9000
```

- spinning up the project (figuring that I have to start the `taiga-db` service **alone** first, otherwise I got some errors with `psycopg2` which cannot connect to postgres)
- visiting `http://my-lan-server.domain.org:9000` -> get redirected to `http://my-lan-server.domain.org:9000/discover`

This page is shown:    
> Taiga
> Something happened and the Taiga has captured the error to be able to work on it.
> Take me home

With these errors in the console:

![image](https://user-images.githubusercontent.com/19967599/106806775-a6713100-6668-11eb-830d-a50f447d8184.png)

Same thing in the network monitor :
![image](https://user-images.githubusercontent.com/19967599/106807242-3911d000-6669-11eb-91e2-71251cd9c6ed.png)

**Expected behavior**
Landing on the normal discover page.

**Configuration**
Ubuntu 18.04 remote server in a LAN.

**Versions**
```
:~$ docker --version
Docker version 20.10.3, build 48d30b5
:~$ docker-compose --version
docker-compose version 1.27.4, build 40524192
```

**Additional context**

I wonder why `localhost:8000` is shown in the browser console messages with method `OPTIONS` because I never used the word `localhost` neither the port `8000` anywhere in the compose files. I actually replaced `localhost` by `my-lan-server.domain.org` and I kept the port `9000` in my environment variable at runtime.

In addition, I cannot figure out where this `conf.json` file located. It's doesn't seem to be in this github repository. Did I missed it?

The strange thing is that when I go to `http://my-lan-server.domain.org:9000/api/v1/` I can see all of the API routes, and that's really great, but I cannot figure out an endpoint for `/api/v1/stats/discover` as printed on the browser console message here-above. 

As the server is in a LAN, it's not yet using HTTPS.

Hope this helps.
And btw, you've done a really nice job with this new release! Many thanks!

**Comment by bameda** [comment]
It seems that `http://my-lan-server.domain.org:9000/conf.json` is a malformed json file and the app is using the default config settings instead of yours.

Can you show me the content of the url http://my-lan-server.domain.org:9000/conf.json

Best regards

**Comment by swiss-knight** [comment]
Woaw, so quick answer! Thanks!

### TL;DR    
I figure out two tiny little details which actually are important;    
1. Boolean used in the compose files are case sensitive regarding of where they are used (either in Python or JS code), **do not use Python boolean style (capitalized)** when they are used in JS code (front-end) and vice versa. This is a bit tricky I think....
2. **Do not comment** the **importers**. If you do not need them; simply **set** their **value** to `false`.

### Details
Here is `conf.json` content within my web browser, but it popped for a few nanoseconds and then the browser displayed this message:
```
SyntaxError: JSON.parse: unexpected character at line 13 column 30 of the JSON data
```

So here's the culprit (`conf.json`):   
```
{
    "api": "http://my-lan-server.domain.org:9000/api/v1/",
    "eventsUrl": "ws://my-lan-server.domain.org:9000/events",
    "eventsMaxMissedHeartbeats": 5,
    "eventsHeartbeatIntervalTime": 60000,
    "eventsReconnectTryInterval": 10000,
    "debug": true,
    "debugInfo": false,
    "defaultLanguage": "en",
    "themes": ["taiga"],
    "defaultTheme": "taiga",
    "defaultLoginEnabled": true,
    "publicRegisterEnabled": True, // <---- line 13
    "feedbackEnabled": true,
    "supportUrl": "https://resources.taiga.io",
    "privacyPolicyUrl": null,
    "termsOfServiceUrl": null,
    "maxUploadFileSize": null,
    "contribPlugins": [
      "/plugins/slack/slack.json",
      "/plugins/github-auth/github-auth.json",
      "/plugins/gitlab-auth/gitlab-auth.json"
    ],
    "gitHubClientId": "",
    "gitLabClientId": "",
    "gitLabUrl": "",
    "tagManager": { "accountId": null },
    "tribeHost": null,
    "enableAsanaImporter": false,
    "enableGithubImporter": , // <---- line 30
    "enableJiraImporter": ,
    "enableTrelloImporter": ,
    "gravatar": false,
    "rtlLanguages": ["fa"]
}
```

It is actually defined as `PUBLIC_REGISTER_ENABLED=True` in a `.env` file which is then directly interpreted by the `docker-compose.yml`.

It seems to be sensitive to the case depending of whether it's used by Python or JS code.... which makes it impossible to define a unique variable if it's used on both the front _and_ the back-end.

To check that, I've tried with `PUBLIC_REGISTER_ENABLED=true` (full lowercase) in my `.env` and it went well (but I guess the backend won't be able to evaluate it correctly anymore now), ...until the next error in the `conf.json` file:
```
SyntaxError: JSON.parse: unexpected character at line 30 column 29 of the JSON data
```
which corresponds to ` "enableGithubImporter": ,` 

I investigated a little, and indeed, I noticed that, as I don't use importers, I had blindly commented them all (`enableGithubImporter`, `enableJiraImporter` and `enableTrelloImporter`). Which resulted in this error.
Instead, if you don't need the importers, you must keep them alive and simply set them to `False`. Both in `taiga-back` and `taiga-front`. This was my mistake.

--- 
I think it could be **nice** for the user not to have to think about the typology-case of input variables, especially booleans. (i.e. let the user insert either "TRUE", "True" or "true", or even any combination of upper and lower-case letters.) and bring all to lowercase in the code.

E.g. if both these lines of `docker-compose.yml` refers to a **single unique** `ENABLE_GITHUB_IMPORTER` env variable: 
   
- https://github.com/taigaio/taiga-docker/blob/e11193b3c604e163c1e71cb37741892d78b7e275/docker-compose.yml#L43     
- https://github.com/taigaio/taiga-docker/blob/e11193b3c604e163c1e71cb37741892d78b7e275/docker-compose.yml#L121

one of the two will not work depending on how it has been defined ("False" vs "false").

Therefore, I strongly recommend letting boolean values be agnostic to the way the user writes them.    
E.g. in `config.py`, instead of:
```
ENABLE_GITHUB_IMPORTER = os.getenv('ENABLE_GITHUB_IMPORTER', 'False') == 'True'
```
Prefer this:
```
ENABLE_GITHUB_IMPORTER = os.getenv('ENABLE_GITHUB_IMPORTER', 'False').lower() == 'true'
```

I didn't check on the front-end side, but the same kind of logic should be implemented (e.g. using [`.toLowerCase()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase) string method) when building the `conf.json` file, for example.




Hope this helps.    
Thanks again for your hint and all your amazing work!

**Comment by bameda** [comment]
Yes, that's the  problem :+1:  Unfortunately your proposal to solve it is not so simple because is incompatible with other settings and  other deployment methods (other than Docker). After thinking a lot about it, we decided to live with the different settings for taiga-front and taiga-back and postpone a refactor that breaks compatibility until later (along with more bigger changes).  

Best regards
